src/index.ts

```diff
@@ -56,11 +56,18 @@ function install(Vue: App, initConf: VueGtmUseOptions = { id: "" }): void {
         if (typeof id === "string") {
           loadScript(id, initConf);
         } else {
-          initConf = {
+          const newConf: VueGtmUseOptions = {
             ...initConf,
-            ...(id.queryParams ?? {}),
           };
-          loadScript(id.id, initConf);
+
+          if (id.queryParams !== null) {
```

Comment:
```suggestion
          if (id.queryParams != null) {
```

Maybe we should also check for `undefined` and not only strict `null`